administration/src/bp-modules/self-service/hooks/useCardGeneratorSelfService.ts

```diff
@@ -31,12 +33,34 @@ type UseCardGeneratorSelfServiceReturn = {
   downloadPdf: (code: CreateCardsResult, fileName: string) => Promise<void>
 }
 
+const handleQueryParams = (
+  cardQueryParams: URLSearchParams
+): {
+  fullName?: string
+  birthday?: PlainDate
+  koblenzReferenceNumber?: string
+} => {
+  const fullName = cardQueryParams.get('name') ?? undefined
+  const birthday = PlainDate.safeFrom(cardQueryParams.get('geburtsdatum')) ?? undefined
+  const koblenzReferenceNumber = cardQueryParams.get('ref') ?? undefined
+  return { fullName, birthday, koblenzReferenceNumber }
+}
+
 const useCardGeneratorSelfService = (): UseCardGeneratorSelfServiceReturn => {
   const projectConfig = useContext(ProjectConfigContext)
   const appToaster = useAppToaster()
+  const [cardQueryParams, setSearchParams] = useSearchParams()
+  const { fullName, birthday, koblenzReferenceNumber } = handleQueryParams(cardQueryParams)
   const [selfServiceCard, setSelfServiceCard] = useState(
-    initializeCard(projectConfig.card, undefined, { expirationDate: null })
+    initializeCard(projectConfig.card, undefined, {
+      fullName,
+      expirationDate: null,
+      extensions: { birthday, koblenzReferenceNumber },
+    })
   )
+  useEffect(() => {
+    setSearchParams(undefined, { replace: true })
+  }, [setSearchParams])
```

Comment:
:upside_down_face: Could you please move this after the `useState` usages and add some newlines around it?